make_and_run_binder.py

```diff
@@ -0,0 +1,155 @@
+    for filename in changes_made.keys():
+        filedata = ""
+        listInd = 0
+        with open(filename, 'r') as fh:
+            for line in fh:
+                if listInd < len(changes_made[filename]) and line == changes_made[filename][listInd][0]:
+                    filedata += changes_made[filename][listInd][1]
+                    listInd += 1
+                else:
+                    filedata += line
+        with open(filename, 'w') as fh:
+            fh.write(filedata)
+    return changes_made
+
+def make_all_includes():
+    # generates an .hpp file with all includes in this project that need to be bound
+    all_includes = []
+    all_include_filename = 'all_cmake_includes.hpp'
+    for filename in (glob.glob(f'{this_project_source}/**/*.hpp', recursive=True) +
+                     glob.glob(f'{this_project_source}/**/*.cpp', recursive=True) +
+                     glob.glob(f'{this_project_source}/**/*.h', recursive=True) +
+                     glob.glob(f'{this_project_source}/**/*.cc', recursive=True) +
+                     glob.glob(f'{this_project_source}/**/*.c', recursive=True)):
+#                     glob.glob(f'{this_project_source}/../build/*/src/*/*.hpp', recursive=True) +  
+#                     glob.glob(f'{this_project_source}/../build/*/src/*/*.cpp', recursive=True) +  
+#                     glob.glob(f'{this_project_source}/../build/*/src/*/*.h', recursive=True) +
+#                     glob.glob(f'{this_project_source}/../build/*/src/*/*.cc', recursive=True) + 
+#                     glob.glob(f'{this_project_source}/../build/*/src/*/*.c', recursive=True)):
+        with open(filename, 'r') as fh:
+            for line in fh:
+                if line.startswith('#include'):
+                    all_includes.append(line.strip())
+    all_includes = list(set(all_includes))
+    # This is to ensure that the list is always the same and doesn't
+    # depend on the filesystem state.  Not technically necessary, but
+    # will cause inconsistent errors without it.
+    all_includes.sort()
+    with open(all_include_filename, 'w') as fh:
+        for include in all_includes:
+            fh.write(f'{include}\n')
+    return all_include_filename
+
+
+def make_bindings_code(all_includes_fn, binder_executable):
+    # runs the binder executable with required parameters
+    shutil.rmtree(bindings_dir, ignore_errors=True)
+    os.mkdir(bindings_dir)
+    # be sure to include dirs for dependencies so it won't get lost
+    proj_include = glob.glob("build/*/src/*/include")
+    proj_include = " -I".join(proj_include)
+    command = (f'{binder_executable} --root-module {python_module_name} '
+               f'--prefix {os.getcwd()}/{bindings_dir}/ '
+               f'--bind {this_project_namespace_to_bind} '
+               + ('--config config.cfg ') +
+               f' {all_includes_fn} -- -std=c++14 '
+               f'-I{this_project_include} {proj_include} -DNDEBUG -v').split()
```

Comment:
It's not a good idea to build a big string and then split here, in case of spaces in arguments. This should just be built as a list to begin with.